From 29cf6d097f12881aa87fa50ee91c1bea562e8ab6 Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Thu, 13 May 2010 19:25:51 -0400
Subject: [PATCH 074/391] immediate-values/immediate-values-fixes-for-modules

Immediate values fixes for modules

Compilation fixes for immediate values when modules are disabled.

Mathieu : merged two fixes.

From: Ingo Molnar <mingo@elte.hu>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@polymtl.ca>
---
 include/linux/immediate.h |    1 -
 include/linux/module.h    |   12 +++++++-----
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/include/linux/immediate.h b/include/linux/immediate.h
index 4791ac0..a74982a 100644
--- a/include/linux/immediate.h
+++ b/include/linux/immediate.h
@@ -63,7 +63,6 @@ extern void imv_unref(struct __imv *begin, struct __imv *end, void *start,
 #define imv_set(name, i)		(name##__imv = (i))
 
 static inline void core_imv_update(void) { }
-static inline void module_imv_update(void) { }
 static inline void imv_unref_core_init(void) { }
 
 #endif
diff --git a/include/linux/module.h b/include/linux/module.h
index 2629fd8..f0990a3 100644
--- a/include/linux/module.h
+++ b/include/linux/module.h
@@ -551,9 +551,6 @@ extern int module_get_iter_markers(struct marker_iter *iter);
 extern void module_update_tracepoints(void);
 extern int module_get_iter_tracepoints(struct tracepoint_iter *iter);
 
-extern void _module_imv_update(void);
-extern void module_imv_update(void);
-
 #else /* !CONFIG_MODULES... */
 #define EXPORT_SYMBOL(sym)
 #define EXPORT_SYMBOL_GPL(sym)
@@ -688,6 +685,12 @@ static inline int module_get_iter_markers(struct marker_iter *iter)
 	return 0;
 }
 
+#endif /* CONFIG_MODULES */
+
+#if defined(CONFIG_MODULES) && defined(CONFIG_IMMEDIATE)
+extern void _module_imv_update(void);
+extern void module_imv_update(void);
+#else
 static inline void _module_imv_update(void)
 {
 }
@@ -695,8 +698,7 @@ static inline void _module_imv_update(void)
 static inline void module_imv_update(void)
 {
 }
-
-#endif /* CONFIG_MODULES */
+#endif
 
 struct device_driver;
 #ifdef CONFIG_SYSFS
-- 
1.6.5.2

